### PR TITLE
(Soul) Underwear Holdout (Soul)

### PR DESCRIPTION
**Pull request by Geevies**
## About The Pull Request

When the ship is in a Red alert or higher status, or is undergoing evacuation, fresh latejoin marines will be provided a self defense kit which they can use to defend themselves until they can gear up properly.

Additionally, marines can now latejoin for up to three minutes after the shuttle crashes into the ship, giving marine players a chance to save their character stuck in cryosleep.

## Why It's Good For The Game

It made absolutely no sense that ARES would just leave marines defenseless when it knows there's a threat incoming or on-board, this serves to close that plot hole by providing latejoiners a fighting chance to survive the threat.

## Changelog

:cl: Geeves, QuickLoad
add: Latejoiners now get a self defense kit if the ship is in red alert or higher, or when evac has been called.
add: Marines can now latejoin for up to three minutes after hijack.
/:cl:

### CODE REVIEW

**Review by stanalbatross — APPROVED** [review]

**Comment by sose96** [comment]
Sounds cool I like it

**Comment by hotimep** [comment]
sounds cool I like it

**Comment by PhantomEpicness** [comment]
UNFATHOMABLY BASED

* Disclaimer *
CM-13 is not related to the Disney owned "Aliens" Franchise and IP This is an entirely fan-made project. In case of an investigation by any federal entity or similar, I do not have any involvement with this group or with the people in it, I do not know how I am here, probably added by a thrid party, I do not support any actions by the member of this group.

**Comment by MusselMain** [comment]
> UNFATHOMABLY BASED
> 
> * Disclaimer *
>   CM-13 is not related to the Disney owned "Aliens" Franchise and IP This is an entirely fan-made project. In case of an investigation by any federal entity or similar, I do not have any involvement with this group or with the people in it, I do not know how I am here, probably added by a thrid party, I do not support any actions by the member of this group.

WY stands for Weston Yamada *ahem* totally not related.

**Comment by AGasCanister** [comment]
so fucking hrp!

**Comment by Nanu308** [comment]
Put Ckey next to the CL at the top, so people can see who added it ingame

File: code/modules/mob/new_player/new_player.dm [hunk 1]
```diff
@@ -244,6 +244,10 @@
 	RoleAuthority.equip_role(character, RoleAuthority.roles_for_mode[rank], T)
 	EquipCustomItems(character)
 
+	if(security_level > SEC_LEVEL_BLUE || EvacuationAuthority.evac_status)
+		to_chat(character, SPAN_HIGHDANGER("As you stagger out of cryogenics, the sleep bay blares: '[EvacuationAuthority.evac_status ? "VESSEL UNDERGOING EVACUATION PROCEDURES, SELF DEFENSE KIT PROVIDED" : "VESSEL IN HEIGHTENED ALERT STATUS, SELF DEFENSE KIT PROVIDED"]'."))
```
Comment (XSlayer300) [inline code/modules/mob/new_player/new_player.dm:248]:
```suggestion
		to_chat(character, SPAN_HIGHDANGER("As you stagger out of hypersleep, the sleep bay blares: '[EvacuationAuthority.evac_status ? "VESSEL UNDERGOING EVACUATION PROCEDURES, SELF DEFENSE KIT PROVIDED" : "VESSEL IN HEIGHTENED ALERT STATUS, SELF DEFENSE KIT PROVIDED"]'."))
```

File: code/modules/mob/new_player/new_player.dm [hunk 2]
```diff
@@ -244,6 +244,10 @@
 	RoleAuthority.equip_role(character, RoleAuthority.roles_for_mode[rank], T)
 	EquipCustomItems(character)
 
+	if(security_level > SEC_LEVEL_BLUE || EvacuationAuthority.evac_status)
+		to_chat(character, SPAN_HIGHDANGER("As you stagger out of hypersleep, the sleep bay blares: '[EvacuationAuthority.evac_status ? "VESSEL UNDERGOING EVACUATION PROCEDURES, SELF DEFENSE KIT PROVIDED" : "VESSEL IN HEIGHTENED ALERT STATUS, SELF DEFENSE KIT PROVIDED"]'."))
+		character.put_in_hands(new /obj/item/storage/box/kit/cryo_self_defense(character.loc))
```
Comment (stanalbatross) [inline code/modules/mob/new_player/new_player.dm:249]: what if they start with two custom items in their hands? make it drop to their loc

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/mob/new_player/new_player.dm (hunk 2 above)
Comment (Chainsawicus) [reply code/modules/mob/new_player/new_player.dm:249]: > what if they start with two custom items in their hands? make it drop to their loc

this is rly embarrassing you kinda got mogged by the code here stan

File: code/modules/projectiles/guns/pistols.dm [hunk 3]
```diff
@@ -331,6 +331,12 @@
 	scatter_unwielded = SCATTER_AMOUNT_TIER_6
 	damage_mult = BASE_BULLET_DAMAGE_MULT
 
+/obj/item/weapon/gun/pistol/holdout/flashlight/handle_starting_attachment()
+	..()
+	var/obj/item/attachable/flashlight/S = new(src)
```
Comment (stanalbatross) [inline code/modules/projectiles/guns/pistols.dm:336]: var name :3

File: code/modules/projectiles/guns/pistols.dm [hunk 4]
```diff
@@ -440,6 +446,13 @@
 /obj/item/weapon/gun/pistol/mod88/training
 	current_mag = /obj/item/ammo_magazine/pistol/mod88/rubber
 
+
+/obj/item/weapon/gun/pistol/mod88/flashlight/handle_starting_attachment()
+	..()
+	var/obj/item/attachable/flashlight/S = new(src)
```
Comment (stanalbatross) [inline code/modules/projectiles/guns/pistols.dm:452]: as above

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by stanalbatross — APPROVED** [review]

**Pull request merged**